feat(components): add helix product typography

sfoster1 opened the pull request:
This PR implements the new helix product typography system. You can find it here: https://www.figma.com/design/1ot18My22DALIcjdLl5LJh/Helix-Design-System?node-id=3571-46578&t=B4pPbtL9mapki2Jg-4

The Product type system is distinct from at least the Web type system, which will have different styles for readability in a different context. It is currently only defined for places other than the ODD.

This ended up being a big PR because our usage of the `StyledText` atom had to be changed somewhat.

The change was necessary because the helix product styles do not and are not intended to align with the semantic styling presented by the `StyledText` component - there is no direct alignment that "when you're making a `<p>`, you use this style". Instead, the intent is that higher level components will use the style by the name it has in helix, without an explicit semantic link outside of the designs. That makes the way the old styled text works kind of nonsense.

In addition, there are no helix product styles for the ODD, and there is no current map and possibly no intention to have a future map between the semantic styles of the ODD and Desktop. That means that having a `StyledText` component that maps from a single semantic element to a set of appropriate styles just doesn't really make sense.

So, `StyledText` is renamed to `LegacyStyledText` including at all callsites. The new `StyledText` requires you to specify styles separately for ODD and Desktop, though it gives you a handy type system integrated hint towards which styles should be allowed. Well, not quite required, more like you have no way to specify both in one prop.

 You can view it commit by commit:

- https://github.com/Opentrons/opentrons/pull/15459/commits/07dae4b338d5b0966fc186fa27a944005a2e3fda adds the actual data for the helix product typography styling, and stories for viewing them as design components (and for viewing the old typography styles). This is more or less the only commit there was during the initial review, though it didn't then have the stories for the old typography
- https://github.com/Opentrons/opentrons/pull/15459/commits/a36f50178c93d95f9cf86610fe8313ae643afdaa is a repo-wide find and replace to change `StyledText` to `LegacyStyledText`
- https://github.com/Opentrons/opentrons/pull/15459/commits/acec9340bcc955921c5007eb388ffec06ce6ac3c implements the new `StyledText`.

Stories here: https://s3-us-west-2.amazonaws.com/opentrons-components/EXEC-566-add-helix-product-typography/index.html?path=/docs/design-tokens-typography--docs

Closes EXEC-566

sfoster1 committed:
feat(components): add helix product typography

- Adds the new sizes for type in the helix product system
- Adds a dummy component under "Design Tokens" in storybook so you can
  view all the text

Closes EXEC-566

Make the storybook display all the stories

koji commented on components/src/helix-design-system/product/typography.ts:
```diff
@@ -0,0 +1,137 @@
+export const fontSizeDisplayBold = fontSizeDisplay
+export const lineHeightDisplayBold = lineHeightDisplay
+export const fontFamilyDisplayBold = fontFamilyDisplay
+export const fontWeightDisplayBold = '700'
```
maybe define font-weight ?

```ts
const regular = 400'
const semiBold = '600'
const bold = '700'
```

koji commented on components/src/helix-design-system/product/typography.ts:
```diff
@@ -0,0 +1,137 @@
+export const fontSizeCaptionMedium = fontSizeCaption
+export const lineHeightCaptionMedium = lineHeightCaption
+export const fontFamilyCaptionMedium = fontFamilyCaption
+export const fontWeightCaptionMedium = '500px'
```
500?

sfoster1 replied:
yup, direct per @mmencarelli . Caption's faded slightly.

sfoster1 replied:
whoops

koji commented on components/src/helix-design-system/product/typography.ts:
```diff
@@ -0,0 +1,137 @@
+export const fontSizeCaptionRegular = fontSizeCaption
+export const lineHeightCaptionRegular = lineHeightCaption
+export const fontFamilyCaptionRegular = fontFamilyCaption
+export const fontWeightCaptionRegular = '400px'
```
400?

sfoster1 replied:
same as above, direct from design

sfoster1 replied:
Whoops

koji reviewed (approved):
lgtm!
thank you for adding this!

sfoster1 committed:
s/StyledText/LegacyStyledText

LegacyStyledText will use the current styles; a new StyledText
implementation will use the helix design system typography.

sfoster1 committed:
feat(components): StyledText styles for app/odd

StyledText is a component that is only really useful if you're goign to
do semantic styles. Semantic styles are only really useful if they mean
the same thing in all the places they're used. This isn't really true
right now, but we can at least have enumerated style tokens that you can
select from when creating one.

Now, which style token you use is separately selectable across ODD and
non-ODD flows. You can still fully or partially override the CSS if you
want, but the styles on desktop align with helix and the styles on ODD
align with what's going on in ODD.

koji commented on components/src/helix-design-system/product/typography.ts:
```diff
@@ -0,0 +1,141 @@
+ */
+
+// Valid for most typography styles
+export const fontFamily = 'Public Sans'
```
can we use this in `app/src/atoms/GlobalStyle/index.ts`?

koji commented on components/src/atoms/StyledText/index.tsx:
use `.ts` itead of `.tsx`?

koji commented on components/src/helix-design-system/product/typography.ts:
```diff
@@ -0,0 +1,141 @@
+// Code
+const fontSizeCode = '0.75rem' // 12px
+const lineHeightCode = '1.25rem' // 20px
+const fontFamilyCode = 'Reddit Mono'
```
can we use `Reddit Mono` without installing a package?

sfoster1 replied:
Definitely not, or at least without installing a font however we install a font - if you look at storybook, this shows up as whatever the browser fallback is. I really haven't investigated how/whether we can isntall this, so I think that if we don't want to keep it as a placeholder I'd rather delete it for now.

koji replied:
For Public Sans, `app` is using `@fontsource` and fontsource supports `reddit mono`.

https://github.com/fontsource/font-files/blob/025ea945ff916c7bfe4fe5aaef1f3f57a523a1c6/fonts/google/reddit-mono/README.md?plain=1

koji commented on components/src/helix-design-system/product/typography.ts:
```diff
@@ -0,0 +1,141 @@
+// Caption
+const fontSizeCaption = '0.8125rem' // 13px
+const lineHeightCaption = '1rem' // 16px
+const fontFamilyCaption = 'Public Sans'
```
```suggestion
const fontFamilyCaption = 'Public Sans'
```
```suggestion
const fontFamilyCaption = fontFamily
```

koji commented on components/src/helix-design-system/product/typography.ts:
```diff
@@ -0,0 +1,141 @@
+export const fontSizeCaptionRegular = fontSizeCaption
+export const lineHeightCaptionRegular = lineHeightCaption
+export const fontFamilyCaptionRegular = fontFamilyCaption
+export const fontWeightCaptionRegular = '400'
```
```suggestion
export const fontWeightCaptionRegular = fontWeightRegular
```

koji commented on components/src/helix-design-system/product/typography.ts:
```diff
@@ -0,0 +1,141 @@
+export const fontSizeCaptionMedium = fontSizeCaption
+export const lineHeightCaptionMedium = lineHeightCaption
+export const fontFamilyCaptionMedium = fontFamilyCaption
+export const fontWeightCaptionMedium = '500'
```
nit

maybe define as fontWeightMedium and add it to the top fontWeight group?

then

```ts
export const fontWeightCaptionMedium = fontWeightMedium
```

sfoster1 replied:
It's not though, `fontWeightMedium` is 600

sfoster1 replied:
well, `fontWeightSemibold` anyway

sfoster1 replied:
and we got a dqa update to make it 600, so i did this

sfoster1 committed:
review: index.tsx->index.ts

sfoster1 committed:
review: derive global font from helix

sfoster1 committed:
Font weights are now semibold not medium

sfoster1 committed:
review: variable for caption weight

sfoster1 committed:
semibold fixups

sfoster1 committed:
sed fixups

sfoster1 committed:
dqa updates

sfoster1 committed:
lint

sfoster1 committed:
format and lint

sfoster1 committed:
fixups after rebase

jerader reviewed (approved):
very nice 🤩

jerader commented on components/src/atoms/StyledText/StyledText.tsx:
```diff
@@ -0,0 +1,275 @@
+  level1Header: {
+    as: 'h1',
+    style: css`
+      @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
```
question, do we need these media queries if these are exclusively for the ODD?

sfoster1 replied:
Yes because the idea is that you will be passing _both_ a `desktopStyle` and `oddStyle` to the component, so we want them to work together nicely
